File: android/ValleCOM/app/src/main/java/com/valleapp/vallecom/db/DBSugerencias.java
package com.valleapp.vallecom.db;

import android.annotation.SuppressLint;
import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteException;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by valle on 13/10/14.
 */
public class DBSugerencias extends DBBase {


    public String tb_name = "sugerencias";


    public DBSugerencias(Context context) {
        super(context, "sugerencias");
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL("CREATE TABLE IF NOT EXISTS "+ tb_name +" (ID TEXT PRIMARY KEY, IDTecla TEXT, sugerencia TEXT )");
    }

    @SuppressLint("Range")
    @Override
    protected JSONObject cursorToJSON(Cursor res) {
        JSONObject obj = new JSONObject();
        try {
            obj.put("ID", res.getString(res.getColumnIndex("ID")));
            obj.put("sugerencia", res.getString(res.getColumnIndex("sugerencia")));
            obj.put("tecla", res.getString(res.getColumnIndex("IDTecla")));
        }catch (Exception e){
            e.printStackTrace();
        }
        return obj;
    }

    @Override
    protected ContentValues caragarValues(JSONObject o) {
        ContentValues values = new ContentValues();
        try {
            values.put("ID", o.getInt("id"));
            values.put("IDTecla",o.getString("tecla"));
            values.put("sugerencia",o.getString("sugerencia"));
        }catch (Exception e){
            e.printStackTrace();
        }
        return  values;
    }
    


    public JSONArray getAll()
    {
        return filter(null);
    }


}

File: android/ValleTPV/ValleTPVlib/src/main/java/com/valleapp/valletpvlib/db/DBSugerencias.kt
package com.valleapp.valletpvlib.db

import android.annotation.SuppressLint
import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import org.json.JSONArray
import org.json.JSONObject

class DBSugerencias(context: Context) : DBBase(context, "sugerencias") {


    override fun onCreate(db: SQLiteDatabase) {
         db.execSQL("CREATE TABLE IF NOT EXISTS $tbName (ID TEXT PRIMARY KEY, IDTecla TEXT, sugerencia TEXT, Incremento DECIMAL(10,2));")
    }

    @SuppressLint("Range")
    override fun cursorToJSON(res: Cursor): JSONObject {
        val obj = JSONObject()
        try {
            obj.put("ID", res.getString(res.getColumnIndex("ID")))
            obj.put("sugerencia", res.getString(res.getColumnIndex("sugerencia")))
            obj.put("tecla", res.getString(res.getColumnIndex("IDTecla")))
            obj.put("incremento", res.getDouble(res.getColumnIndex("Incremento")))
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return obj
    }



    override fun cargarValues(o: JSONObject): ContentValues  {
        val values = ContentValues()
        try {
            values.put("ID", o.getInt("id"))
            values.put("IDTecla", o.getString("tecla"))
            values.put("sugerencia", o.getString("sugerencia"))
            values.put("Incremento", o.getDouble("incremento"))
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return values
    }

    fun getAll(): JSONArray {
        return filter(null)
    }

    override fun filter(cWhere: String?): JSONArray {
        val array = JSONArray()
        val db = readableDatabase
        val selectQuery = if (cWhere != null) {
            "SELECT * FROM $tbName WHERE $cWhere ORDER BY Incremento DESC"
        } else {
            "SELECT * FROM $tbName ORDER BY Incremento DESC"
        }

        val cursor = db.rawQuery(selectQuery, null)
        cursor.use {
            if (cursor.moveToFirst()) {
                do {
                    array.put(cursorToJSON(cursor))
                } while (cursor.moveToNext())
            }
        }
        return array
    }
}
